nlmod/gwf/gwf.py

```diff
@@ -499,8 +485,176 @@ def drn(
         return None
 
 
+def riv(
+    ds,
+    gwf,
+    stage="riv_stage",
+    cond="riv_cond",
+    rbot="riv_rbot",
+    pname="riv",
+    auxiliary=None,
+    layer=None,
+    **kwargs,
+):
+    """create river package from model dataset.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        dataset with model data.
+    gwf : flopy ModflowGwf
+        groundwaterflow object.
+    stage : str or xarray.DataArray, optional
+        river stage, either as string pointing to data
+        array in ds or as data array. By default assumes
+        data array is stored under "riv_stage".
+    cond : str or xarray.DataArray, optional
+        river conductance, either as string pointing to data
+        array in ds or as data array. By default assumes
+        data array is stored under "riv_cond".
+    rbot : str or xarray.DataArray, optional
+        river bottom elevation, either as string pointing to data
+        array in ds or as data array. By default assumes
+        data array is stored under "riv_rbot".
+    pname : str, optional
+        package name
+    auxiliary : str or list of str
+        name(s) of data arrays to include as auxiliary data to reclist
+    layer : int or None
+        The layer in which the boundary is added. It is added to the first active layer
+        when layer is None. The default is None.
+
+    Returns
+    -------
+    riv : flopy ModflowGwfriv
+        riv package
+    """
+    logger.info("creating mf6 RIV")
+
+    mask_arr = _get_value_from_ds_datavar(ds, "cond", cond, return_da=True)
+    mask = mask_arr > 0
+
+    first_active_layer = layer is None
+    riv_rec = grid.da_to_reclist(
+        ds,
+        mask=mask,
+        col1=stage,
+        col2=cond,
+        col3=rbot,
+        layer=layer,
+        aux=auxiliary,
+        first_active_layer=first_active_layer,
+        only_active_cells=False,
+    )
+
+    if len(riv_rec) > 0:
+        riv = flopy.mf6.ModflowGwfriv(
+            gwf,
+            maxbound=len(riv_rec),
+            stress_period_data=riv_rec,
+            auxiliary="CONCENTRATION" if auxiliary is not None else None,
+            save_flows=True,
+            pname=pname,
+            **kwargs,
+        )
+        if (auxiliary is not None) and (ds.transport == 1):
+            logger.info("-> adding RIV to SSM sources list")
+            ssm_sources = list(ds.attrs["ssm_sources"])
+            if riv.package_name not in ssm_sources:
+                ssm_sources += [riv.package_name]
+                ds.attrs["ssm_sources"] = ssm_sources
+        return riv
+    else:
+        logger.warning("no riv pkg added")
+        return None
+
+
+def chd(
+    ds,
+    gwf,
+    mask="chd_mask",
+    head="chd_head",
+    pname="chd",
+    auxiliary=None,
+    layer=0,
+    **kwargs,
+):
+    """create constant head boundary at the model's edges from the model dataset.
```

Comment:
This description is wrong I think it's no longer only model edges, but wherever the mask says it should be.